Lib/Protocols/IdFTP.pas

```diff
@@ -2742,18 +2764,45 @@ procedure TIdFTP.IssueFEAT;
       Break;
     end;
   end;
-
+  if FClientInfo.ClientName <> '' then begin
+    if IsExtSupported('CSID') then begin {do not localize}
+      LClnt := 'Name='+FClientInfo.FClientName;

```

Comment:
Rather than having this logic directly in `IssueFEAT`, how about adding a `CsidOutput` property to `TIdFTPClientIdentifier` instead?